v2/anomalydetection/backend-setup.mdx

```diff
@@ -0,0 +1,1781 @@
+            cookiesEnabled: { value: true },
+        },
+    }
+};
+```
+</details>
+
+## Retrieving the Request ID
+
+We pass a `requestId` during the email password sign in, sign up and reset password APIs from the frontend (see [frontend setup section](frontend-setup#generating-a-request-id-on-the-frontend)). You can retrieve this ID by overriding these APIs in the emailpassword recipe on the backend.
+
+Below is an example of how to retrieve the request ID from the input body:
+
+<BackendSDKTabs>
+<TabItem value="nodejs">
+
+```tsx
+    const signInPOST = async function (input: {
+        options: {
+            req: {
+                getJSONBody: () => Promise<{
+                    requestId: string;
+                }>;
+            };
+        };
+    }) {
+        // ... your code here ...
+        const requestId = (await input.options.req.getJSONBody()).requestId;
+        if(!requestId) {
+            return {
+                status: "GENERAL_ERROR",
+                message: "The request ID is required"
+            }
+        }
+        // ... your code here ...
+    }
+```
+
+</TabItem>
+
+<TabItem value="go">
+
+```go
+import (
+	"encoding/json"
+
+	"github.com/supertokens/supertokens-golang/recipe/emailpassword/epmodels"
+	"github.com/supertokens/supertokens-golang/supertokens"
+)
+
+type ReqBody struct {
+	RequestID *string `json:"requestId"`
+}
+
+func signInPOST(formFields []epmodels.TypeFormField, tenantId string, options epmodels.APIOptions, userContext supertokens.UserContext) (epmodels.SignInPOSTResponse, error) {
+    // ... your code here ...
+    var reqBody ReqBody
+    err := json.NewDecoder(options.Req.Body).Decode(&reqBody)
+    if err != nil {
+        return epmodels.SignInPOSTResponse{}, err
+    }
+
+    if reqBody.RequestID == nil {
+        return epmodels.SignInPOSTResponse{
+            GeneralError: &supertokens.GeneralErrorResponse{
+                Message: "The request ID is required",
+            },
+        }, nil
+    }
+    requestId := *reqBody.RequestID
+    _ = requestId
+    // ... your code here ...
+
+    return epmodels.SignInPOSTResponse{}, nil
+}
+```
+
+</TabItem>
+
+<TabItem value="python">
+
+```python
+from supertokens_python.recipe.emailpassword.interfaces import APIOptions
+from supertokens_python.types import GeneralErrorResponse
+
+async def sign_in_post(api_options: APIOptions):
+    # ... your code here ...
+    request_body = await api_options.request.json()
+    if not request_body:
+        return GeneralErrorResponse(message="The request body is required")
+
+    request_id = request_body.get("requestId")
+    if not request_id:
+        return GeneralErrorResponse(message="The request ID is required")
+    # ... your code here ...
+```
+
+</TabItem>
+
+<TabItem value="otherFrameworks">
+
+:::note
+For using with other frameworks, you can follow the same steps as above to override the APIs and add custom logic for anomaly detection.
```

Comment:
There is no "as above" from the user's point of view, cause this is a tab layout. Also, the bullet points are shown if this is selected as well, which is misleading, cause the bullet points are for explaining the code snippets.

Reply:
Same applies to the passwordless section one

Reply:
done